content/docs/guides/neon-authorize-drizzle.md

```diff
@@ -0,0 +1,199 @@
+subtitle: Use Drizzle crudPolicy to manage Row-Level Security with Neon Authorize
+enableTableOfContents: true
+---
+
+<InfoBlock>
+<DocsList title="What you'll learn">
+<p>How to simplify Row-Level Security using crudPolicy</p>
+<p>Common RLS patterns with Drizzle</p>
+</DocsList>
+
+<DocsList title="Related docs" theme="docs">
+  <a href="/docs/guides/neon-authorize">About Neon Authorize</a>
+  <a href="https://orm.drizzle.team/docs/rls">RLS in Drizzle</a>
+</DocsList>
+
+</InfoBlock>
+
+## Why simplify RLS policies?
+
+Row-Level Security (RLS) is an important last line of defense for protecting your data at the database level. However, implementing RLS requires writing and maintaining separate SQL policies for each CRUD operation (Create, Read, Update, Delete), which can be both tedious and error-prone.
+
+### For example
+
+To illustrate, let's consider a simple **Todo** list app with RLS policies applied to a `todos` table. Postgres RLS policies use two types of conditions:
+
+- `USING` clause — controls which existing rows can be accessed
+- `WITH CHECK` clause — controls what new or modified data can be written
+
+Here's how these clauses apply to each operation:
+
+| Operation | USING clause               | WITH CHECK clause          |
+| --------- | -------------------------- | -------------------------- |
+| Select    | `auth.user_id() = user_id` |                            |
+| Insert    |                            | `auth.user_id() = user_id` |
+| Update    | `auth.user_id() = user_id` | `auth.user_id() = user_id` |
+| Delete    | `auth.user_id() = user_id` |                            |
+
+And the SQL code would look like this:
+
+```sql shouldWrap
+CREATE TABLE IF NOT EXISTS "todos" (
+    "id" bigint PRIMARY KEY,
+    "user_id" text DEFAULT (auth.user_id()) NOT NULL,
+    "task" text NOT NULL,
+    "is_complete" boolean DEFAULT false NOT NULL,
+    "inserted_at" timestamp with time zone DEFAULT now() NOT NULL
+);
+
+ALTER TABLE "todos" ENABLE ROW LEVEL SECURITY;
+
+CREATE POLICY "create todos" ON "todos" AS PERMISSIVE FOR INSERT TO "authenticated" WITH CHECK ((select auth.user_id() = user_id));
+
+CREATE POLICY "view todos" ON "todos" AS PERMISSIVE FOR SELECT TO "authenticated" USING ((select auth.user_id() = user_id));
+
+CREATE POLICY "crud-authenticated-policy-update" ON "todos" AS PERMISSIVE FOR UPDATE TO "authenticated" USING ((select auth.user_id() = "todos"."user_id")) WITH CHECK ((select auth.user_id() = "todos"."user_id"));
+
+CREATE POLICY "delete todos" ON "todos" AS PERMISSIVE FOR DELETE TO "authenticated" USING ((select auth.user_id() = user_id));
+```
+
+As you add new features, you'll need to add more policies to match. This growing complexity can lead to subtle bugs that can be hard to spot in a large schema file filled with SQL statements.
+
+## Simplifying RLS with crudPolicy
+
+The `crudPolicy` function generates RLS policies by accepting a simple configuration object. Let's break down its usage:
+
```

Comment:
maybe highlight the relevant lines in the code example?

Reply:
Done.